src/legilogic_ethereum/ethereum_user.ml

```diff
@@ -637,30 +646,31 @@ let post_operation_kernel : Ethereum_chain.Operation.t -> Address.t -> TokenAmou
      fail error (* bork "Cannot match this" *)
   | FinalTransactionStatus.Confirmed (_transaction, _signed, receipt) ->
      if ethereum_user_log then
-       Logging.log "transaction status=%s" (print_status_receipt receipt);
+       Logging.log "transaction status=%B" (get_status_receipt receipt);
      Lwt_exn.return receipt))
 
-let post_operation : Ethereum_chain.Operation.t
-                          -> Address.t
-                          -> TokenAmount.t
-                          -> TransactionReceipt.t Lwt_exn.t =
-  fun operation sender value ->
+let post_operation : operation:Ethereum_chain.Operation.t -> sender:Address.t -> value:(TokenAmount.t option) -> TransactionReceipt.t Lwt_exn.t =
+  fun ~operation ~sender ~value ->
   let rec submit_operation : unit -> TransactionReceipt.t Lwt_exn.t =
     fun () ->
     Lwt_exn.bind (post_operation_kernel operation sender value)
       (fun ereceipt ->
-        (let str = print_status_receipt ereceipt in
-         let str_succ = "1" in
-         if String.equal str str_succ then
+        (if get_status_receipt ereceipt then
            Lwt_exn.return ereceipt
          else
            (if ethereum_user_log then
-              Logging.log "receipt is not 1, ereceipt=%s" str;
+              Logging.log "receipt is not true, ereceipt=%s" (TokenAmount.to_string ereceipt.status);
             Lwt_exn.bind (Ethereum_watch.sleep_delay_exn 1.0) (fun () -> submit_operation ()))
         )
       ) in
   submit_operation ()
 
+let option_resolution : 'a -> 'a option -> 'a =
```

Comment:
There should be a function in `Lib.Option` for that already. If not, move it there.

Reply:
I would like if that were possible. The "Option" is described in http://ocaml-lib.sourceforge.net/doc/Option.html

What we have is the functionality.
val default : 'a -> 'a option -> 'a

Unfortunately, the "Option.default" is not working adequately to my surprise. And so I used this way. Any help for that is much welcomed.